File: app/src/main/java/com/saienko/androidthings/barman/db/cocktail/CocktailDao.java
package com.saienko.androidthings.barman.db.cocktail;

import android.arch.persistence.room.*;

import java.util.List;

/**
 * Created
 * User: Vasiliy Saienko
 * Date: 9/9/17
 * Time: 14:08
 */

@Dao
public interface CocktailDao {
    @Query("SELECT * FROM cocktail")
    List<Cocktail> getAll();

    @Insert
    void insertAll(Cocktail... drinks);

    @Insert
    long insert(Cocktail drink);

    @Delete
    void delete(Cocktail drink);

    @Query("DELETE FROM cocktail WHERE cocktailId = :cocktailId")
    void delete(long cocktailId);

    @Query("DELETE FROM cocktail")
    void clear();

    @Update
    void update(Cocktail cocktail);

    @Query("SELECT * FROM cocktail WHERE cocktailGroupId = :cocktailGroupId")
    List<Cocktail> getByGroup(long cocktailGroupId);

    @Query("SELECT * FROM cocktail WHERE cocktailName LIKE :cocktailName")
    Cocktail getByName(String cocktailName);

//    @Query("DELETE  FROM cocktail WHERE cocktailId = :cocktailId")
//    void deleteById(long cocktailId);
}

File: app/src/main/java/com/saienko/androidthings/barman/db/dao/CocktailDao.kt
package com.saienko.androidthings.barman.db.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import com.saienko.androidthings.barman.db.entity.Cocktail

/**
 * Created
 * User: Vasiliy Saienko
 * Date: 9/9/17
 * Time: 14:08
 */

@Dao
interface CocktailDao : BaseDao<Cocktail> {
    @get:Query("SELECT * FROM cocktail")
    val all: MutableList<Cocktail>

    @Insert
    fun insertAll(vararg cocktails: Cocktail)

    @Query("DELETE FROM cocktail WHERE id = :cocktailId")
    fun delete(cocktailId: Long)

    @Query("DELETE FROM cocktail")
    fun clear()

    @Query("SELECT * FROM cocktail WHERE id LIKE :cocktailId")
    fun get(cocktailId: Long): Cocktail

}
